extensions/extensions-pipeline-invocation-service/src/main/java/com/boozallen/aissemble/pipeline/invocation/service/endpoint/MessageEndpoint.java

```diff
@@ -38,13 +41,26 @@ public class MessageEndpoint {
     @Incoming(CHANNEL_NAME)
     @Acknowledgment(Acknowledgment.Strategy.PRE_PROCESSING)
     public void receivePipelineRequest(String rawRequest) {
-        try {
-            PipelineInvocationRequest pipelineInvocationRequest = PipelineInvocationRequest.fromString(rawRequest);
-            logger.info("Received message request to submit " + pipelineInvocationRequest.getApplicationName() + ".");
-            pipelineInvocationAgent.submitSparkApplication(pipelineInvocationRequest);
-            logger.info("Submitted " + pipelineInvocationRequest.getApplicationName() + " for processing.");
-        } catch (Exception failsafe) {
-            throw new RuntimeException(failsafe);
-        }
+        // Running the helm command asynchronously in a separated thread to avoid vertx event loop thread block issue
+        ExecutorService executorService = Executors.newFixedThreadPool(1);
```

Comment:
A: Since the expected throughput of this service is low and the helm command time is dominated by I/O wait this isn't as big of a deal as it usually would be, but we should still be sharing a single thread pool with a reasonable number of threads.  If we did just want to create background threads unbounded, it would be best not to create a pool at all.

Reply:
Per discussion, updated the thread pool number and use the quarkus showdown() to shutdown the ExecutorService